doc/users/interactive_guide.rst

```diff
@@ -0,0 +1,432 @@
+Draw Idle
+=========
+
+.. autosummary::
+   :template: autosummary.rst
+   :nosignatures:
+
+   backend_bases.FigureCanvasBase.draw
+   backend_bases.FigureCanvasBase.draw_idle
+   backend_bases.FigureCanvasBase.flush_events
+
+
+In almost all cases, we recommend using
+`backend_bases.FigureCanvasBase.draw_idle` over
+`backend_bases.FigureCanvasBase.draw`.  ``draw`` forces a rendering of
+the figure where as ``draw_idle`` schedules a rendering the next time
+the GUI window is going to re-paint the screen.  This improves
+performance by only rendering pixels that will be shown on the screen.  If
+you want to be sure that the screen is updated as soon as possible do ::
+
+  fig.canvas.draw_idle()
+  fig.canvas.flush_events()
+
+
+
+Threading
+=========
+
+Most GUI frameworks require that all updates to the screen, and hence
+their main event loop, run on the main thread.  This makes pushing
+periodic updates of a plot to a background thread impossible.
+Although it seems backwards, it is typically easier to push your
+computations to a background thread and periodically update
+the figure on the main thread.
+
+In general Matplotlib is not thread safe.  If you are going to update
+`.Artist` objects in one thread and draw from another you should make
+sure that you are locking in the critical sections.
+
+
+The Weeds
+=========
+
+
+Eventloop integration mechanism
+-------------------------------
+
+CPython / readline
+~~~~~~~~~~~~~~~~~~
+
+The python capi provides a hook, :c:var:`PyOS_InputHook`, to register a
+function to be run "The function will be called when Python's
+interpreter prompt is about to become idle and wait for user input
+from the terminal.".  This hook can be used to integrate a second
+event loop (the GUI event loop) with the python input prompt loop.
+The hook functions typically exhaust all pending events on the GUI
+event queue, run the main loop for a short fixed amount of time, or
+run the event loop until a key is pressed on stdin.
+
+
+Matplotlib does not currently do any management of
+:c:var:`PyOS_InputHook` due to the wide range of ways that matplotlib
+is used.  This management is left to downstream libraries -- either
+user code or the shell.  Interactive figures, even with matplotlib in
+'interactive mode', may not work in the vanilla python repl if an
+appropriate :c:var:`PyOS_InputHook` is not registered.
+
+Input hooks, and helpers to install them, are usually included with
+the python bindings for GUI toolkits and may be registered on import.
+IPython also ships input hook functions for all of the GUI frameworks
+Matplotlib supports which can be installed via ``%matplotlib``.  This
+is the recommended method of integrating Matplotlib and a prompt.
+
+
+IPython / prompt toolkit
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+With IPython >= 5.0 IPython has changed from using cpython's readline
+based prompt to a ``prompt_toolkit`` based prompt.  ``prompt_toolkit``
+has the same conceptual input hook, which is feed into pt via the
```

Comment:
pt = ptk or a typo for "it"?

Reply:
I thought prompttoolkit appreviated it self 'pt', but I will write it out anyway.